### ISSUE TITLE
Transaction not retrying after upgrade from 3.7.5 to 4.2.0

### ISSUE THREAD

**Issue by novemberborn** [googleapis/nodejs-firestore#1302]
After upgrading from 3.7.5 to 4.2.0 I'm seeing transactions fail with a code 10 error:

> 10 ABORTED: Aborted due to cross-transaction contention. This occurs when multiple transactions attempt to access the same data, requiring Firestore to abort at least one in order to enforce serializability.

The contention itself is expected. The transaction reads several documents and possibly writes them as well. There are concurrent requests to the service which can cause concurrent transactions. According to my instrumentation with 3.7.5 the transaction function was retried. That's seemingly not the case with 4.2.0.

The error occurs when reading the documents, not when writing.

From what I can tell we do get an error with the right code, and it does propagate up the call stack. Which means it should retry. I can't really tell why that's not the case. Could it be that either `begin()` or `rollback()` are throwing a similar error?

https://github.com/googleapis/nodejs-firestore/blob/786e52f8c8b7b9c6b84ffc988190470a063d5855/dev/src/transaction.ts#L438

https://github.com/googleapis/nodejs-firestore/blob/786e52f8c8b7b9c6b84ffc988190470a063d5855/dev/src/transaction.ts#L458

#### Environment details

  - OS: https://hub.docker.com/layers/node/library/node/12.18.4-slim/images/sha256-920108f9cfbb7882f0484a85e21a6c8a677f6c8a4ca0fba0020a76807931d988?context=explore
  - Node.js version: 12.18.4
  - npm version: 6.14.8
  - `@google-cloud/firestore` version: 4.2.0

**Comment by schmidt-sebastian** [comment]
@novemberborn Thanks for filing this. Do you happen to have logs from one of your transaction attempts? You can obtain them by invoking `setLogFunction(console.log)`.

**Comment by novemberborn** [comment]
> Do you happen to have logs from one of your transaction attempts? You can obtain them by invoking `setLogFunction(console.log)`.

I will see how best to set this up in our service and get back to you.

**Comment by novemberborn** [comment]
So this is embarrassing.

Our transaction code is wrapped in some other utility code, which was upgraded at the same time. I thought I ruled out there being any bugs there but as it turns out it was rethrowing the original (first) exception.

Adding the logging, and looking at our instrumentation, showed that the SDK was retrying. And then I looked more closely and spotted the bug. All's good now. And at least I managed to fix a bug with `setLogFunction(null)` 😉 

Thanks for the help and the work on this library.
